tools/vsextension/ProjectWizard/VCTargets/OpenEnclave.Cpp.Clang.targets

```diff
@@ -0,0 +1,295 @@
+                  ThumbMode="%(ClCompile.ThumbMode)"
+                  TreatWarningAsError="%(ClCompile.TreatWarningAsError)"
+                  UndefineAllPreprocessorDefinitions="%(ClCompile.UndefineAllPreprocessorDefinitions)"
+                  UndefinePreprocessorDefinitions="%(ClCompile.UndefinePreprocessorDefinitions)"
+                  UseShortEnums="%(ClCompile.UseShortEnums)"
+                  Verbose="%(ClCompile.Verbose)"
+                  WarningLevel="%(ClCompile.WarningLevel)"
+
+                  GccToolChain="%(ClCompile.GccToolChain)"
+                  Sysroot="%(ClCompile.Sysroot)"
+                  TargetArch="%(ClCompile.TargetArch)"
+
+                  TrackFileAccess="$(TrackFileAccess)"
+                  TrackerLogDirectory="$(TLogLocation)"
+                  MinimalRebuildFromTracking="%(ClCompile.MinimalRebuildFromTracking)"
+                  TLogReadFiles="@(CLTLogReadFiles)"
+                  TLogWriteFiles="@(CLTLogWriteFiles)"
+                  ToolExe="$(ClangToolExe)"
+                  ToolPath="$(ClangToolPath)"
+                  ToolArchitecture="$(CLToolArchitecture)"
+                  TrackerFrameworkPath="$(CLTrackerFrameworkPath)"
+                  TrackerSdkPath="$(CLTrackerSdkPath)"
+                  EnableExecuteTool="$(ClangEnableExecuteTool)"
+                  MSExtensions="%(ClCompile.MSExtensions)"
+                  MSCompatibility="%(ClCompile.MSCompatibility)"
+                  MSCompatibilityVersion="%(ClCompile.MSCompatibilityVersion)"
+                  MSCompilerVersion="%(ClCompile.MSCompilerVersion)"
+    >
+    </ClangCompile>
+  </Target>
+
+  <Target Name="Link" Condition="'@(Link)' != ''">
+    <PropertyGroup>
+      <ObjectsList>@(Link)</ObjectsList>
+      <ObjectsListSpaced>$(ObjectsList.Replace(';', ' '))</ObjectsListSpaced>
+      <LibsList>;%(Link.AdditionalDependencies)</LibsList>
+      <LibsListSpaced>$(LibsList.TrimEnd({';').Replace(';', ' -l'))</LibsListSpaced>
+      <OutDirTrimmed>$(OutDir.TrimEnd({'\\'))</OutDirTrimmed>
+    </PropertyGroup>
+
+    <Exec Command='"$(ClangToolExe)" -target x86_64-pc-linux $(ObjectsListSpaced) -o "$(OutDir)$(TargetName).elf" -nostdlib -nodefaultlibs -nostartfiles -Wl,--no-undefined,-Bstatic,-Bsymbolic,--export-dynamic,-pie,--build-id -Wl,-z,noexecstack -Wl,-z,now -Wl,-gc-sections -fuse-ld=lld %(Link.AdditionalOptions) "-L$(OutDirTrimmed)" $(LibsListSpaced) -loeenclave -loelibcxx -loelibc -loecryptombed -lmbedtls -lmbedx509 -lmbedcrypto -loelibc -loesyscall -loecore'>
+    </Exec>
+  </Target>
+
+  <Target Name="Lib" Condition="'@(Lib)' != ''"
+          Inputs="@(Lib)"
+          Outputs="$(OutDir)lib$(TargetName).a">
+    <PropertyGroup>
+      <ObjectsList>@(Lib)</ObjectsList>
+      <ObjectsListSpaced>$(ObjectsList.Replace(';', ' '))</ObjectsListSpaced>
+    </PropertyGroup>
+    <Exec Command='"$(LlvmArExecutable)" rcs %(Lib.AdditionalOptions) "$(OutDir)lib$(TargetName).a" $(ObjectsListSpaced)'>
```

Comment:
Great work to enable ELF enclave building via VS using clang cross-compilation!

Reply:
Thanks!